src/ripple/rpc/impl/RPCHandler.cpp

```diff
@@ -157,6 +157,12 @@ error_code_i fillHandler (Context& context,
         return rpcNO_NETWORK;
     }
 
+    if (context.app.getOPs().isAmendmentBlocked() &&
+        handler->condition_ & NEEDS_CURRENT_LEDGER)
+    {
+        return rpcAMENDMENT_BLOCKED;
+    }
+
     if (!context.app.config().standalone() &&
         handler->condition_ & NEEDS_CURRENT_LEDGER)
```

Comment:
Should the `NEEDS_CLOSED_LEDGER` also only work if not amendment blocked? Note that I don't see any RPCs actually using that condition.